pkg/restapi/issuer/operation/operations.go

```diff
@@ -1452,6 +1523,62 @@ func (o *Operation) handleWACIRequestCredential(msg service.DIDCommAction, profi
 	return issuecredential.WithIssueCredential(issueCredMsg), nil
 }
 
+func getAttachments(action service.DIDCommAction) ([]decorator.GenericAttachment, error) {
+	var didCommMsgAsMap map[string]interface{}
+
+	err := action.Message.Decode(&didCommMsgAsMap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode didComm action message: %w", err)
+	}
+
+	attachmentsRaw, ok := didCommMsgAsMap["requests~attach"]
+	if !ok {
+		return nil, errors.New("missing attachments from DIDComm message map")
+	}
+
+	attachmentsAsArrayOfInterfaces, ok := attachmentsRaw.([]interface{})
+	if !ok {
+		return nil, errors.New("attachments were not an array of interfaces as expected")
+	}
+
+	attachmentsBytes, err := json.Marshal(attachmentsAsArrayOfInterfaces)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal attachments as array of interfaces: %w", err)
+	}
+
+	var attachments []decorator.GenericAttachment
+
+	err = json.Unmarshal(attachmentsBytes, &attachments)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshall generic attachments: %w", err)
+	}
+
+	return attachments, nil
+}
+
+func getCredentialApplicationFromAttachment(attachment *decorator.GenericAttachment) ([]byte, error) {
+	if attachment.Data.JSON != nil {
+		attachmentAsMap, ok := attachment.Data.JSON.(map[string]interface{})
+		if !ok {
+			return nil, errors.New("couldn't assert attachment as a map")
+		}
+
+		credentialApplicationRaw, ok := attachmentAsMap["credential_application"]
+		if !ok {
+			return nil, errors.New("credential_application object missing from attachment")
+		}
+
+		credentialApplicationBytes, err := json.Marshal(credentialApplicationRaw)
+		if err != nil {
+			return nil, fmt.Errorf("failed to marshal credential application: %w", err)
+		}
+
+		return credentialApplicationBytes, nil
+	}
+
+	return nil, nil
```

Comment:
It seems to me like you should be returning an error here if `attachment.Data.JSON == nil`

(also note that, in general, when you have a function returning some value + an error, you usually want to ensure that either the value is nil or the error is nil, but not both unless there's a specific reason for it)

Reply:
(in other words, you generally don't see `return nil, nil`

Reply:
The reason i have err as nil  is we are only reading credential application if found other wise flow will continue . @DRK3